src/main/java/spotify/factories/AbstractPlayableObjectDeserializer.java

```diff
@@ -7,13 +7,46 @@
 
 import java.lang.reflect.Type;
 
+/**
+ * This is a AbstractPlayableObjectDeserializer class to implement JsonDeserializer.
+ *
+ * @see spotify.models.generic.AbstractPlayableObject
+ * @author Jiankai Zheng (jk.zheng@hotmail.com)
+ * @since // check with Jiankai
+ */
+
 class AbstractPlayableObjectDeserializer implements JsonDeserializer<AbstractPlayableObject> {
+
+    /**
+     * Custom deserialize method to deserialize json, type and jsonDeserializationContext
```

Comment:
This doesn't really also make it any clearer, it should mention something about that it deserializes based on the type returned in the json

Reply:
Would the following be better? Thanks.

Deserialize json, type and jsonDeserializationContext and return childClass which either contains track or episode

Reply:
No, this is just the same but the order of words are just rearranged. The description should tell in simple words what the function is doing.

Reply:
Thanks. I will try again and will include the updated one in the next PR.